pydantic_ai_slim/pydantic_ai/models/openai.py

```
@@ -502,16 +503,59 @@ class _ResponsesRequestParams:
+    return request_options
+
+
+def _add_openai_prompt_cache_breakpoint(
+    content: Sequence[ChatCompletionContentPartParam | responses.ResponseInputContentParam],
+) -> None:
+    if not content:
+        raise UserError(
+            '`CachePoint` cannot be the first item in an OpenAI user prompt; '
+            'it must follow content to attach the cache breakpoint to.'
+        )
+    breakpoint: _OpenAIPromptCacheBreakpoint = {'mode': 'explicit'}
```

Ruff (0.15.20)

[error] Variable `breakpoint` is shadowing a Python builtin

(A001)